inference-engine/src/mkldnn_plugin/nodes/mkldnn_deconv_node.cpp

```diff
@@ -334,94 +407,300 @@ bool MKLDNNDeconvolutionNode::created() const {
+        descs.emplace_back(createDescriptorInternalInt8(in_candidate, wgh_candidate, out_candidate));
     } else {
-        mkldnn::memory::desc wgh_candidate(MKLDNNExtensionUtils::convertToDnnlDims(weightDims), in_candidate.getDataType(), memory::format_tag::any);
+        mkldnn::memory::desc wgh_candidate(MKLDNNExtensionUtils::convertToDnnlDims(getWeightDims()),
+                                           dnnlInDesc.getDataType(), memory::format_tag::any);
         for (auto alg : {mkldnn::algorithm::convolution_winograd, mkldnn::algorithm::convolution_direct}) {
-            std::shared_ptr<mkldnn::convolution_forward::desc> conv_desc;
-            conv_desc.reset(new convolution_forward::desc(prop_kind::forward_inference, alg,
-                                                          out_candidate.getDnnlDesc(), wgh_candidate, in_candidate.getDnnlDesc(),
-                                                          convertDims(stride),
-                                                          convertDims(dilation),
-                                                          convertDims(paddingL),
-                                                          convertDims(paddingR)));
-
-            std::shared_ptr<mkldnn::convolution_backward_data::desc> deconv_desc;
-            deconv_desc.reset(new convolution_backward_data::desc(alg, out_candidate.getDnnlDesc(), wgh_candidate,
-                                                                  in_candidate.getDnnlDesc(),
-                                                                  convertDims(stride),
-                                                                  convertDims(dilation),
-                                                                  convertDims(paddingL),
-                                                                  convertDims(paddingR)));
-            descs_fwd.push_back(conv_desc);
-            descs_bwd.push_back(deconv_desc);
-
-            auto fwd_conv_pd = std::make_shared<convolution_forward::primitive_desc>(*conv_desc, getEngine(), true);
+            std::shared_ptr<convolution_backward_data::desc> deconv_desc;
+            std::shared_ptr<convolution_forward::primitive_desc> fwd_conv_pd;
+            std::tie(deconv_desc, fwd_conv_pd) = createDescriptorInternalDefault(in_candidate, wgh_candidate, out_candidate, alg);
             if (fwd_conv_pd->get(true) == nullptr)
                 continue;
-
             descs.emplace_back(deconv_desc, fwd_conv_pd);
         }
     }
 }
 
 std::shared_ptr<MemoryDesc> MKLDNNDeconvolutionNode::getSrcMemDesc(mkldnn::primitive_desc_iterator &primitive_desc_it, size_t idx) {
-    if (idx == 2) {
-        return std::make_shared<CpuBlockedMemoryDesc>(getOriginalInputPrecisionAtPort(2), Shape(getInputShapeAtPort(2).getStaticDims()));
+    if (idx > 0) {
+        if (isInt8) {
```

Comment:
Could you please ellaborate why we need this condition?

Reply:
At this moment we also save weight as second input for INT8 deconv. We need this because it's not easy to avoid current "hack" when we save weight data as is, but with different order and reorder it after this.
This condition necessary to avoid inserting additional reorder between weight and deconvolution.

Reply:
So I expect to see this comment in the code

Reply:
Also you have `if (idx == 2)` condition under `if (idx > 0)`. Seem like it shouldn't be nested.

Reply:
fixed